src/gui/addnewtorrentdialog.cpp

```diff
@@ -118,6 +119,50 @@ namespace
         const BitTorrent::TorrentInfo &m_torrentInfo;
         QStringList &m_filePaths;
     };
+
+    // savePath is a folder, not an absolute file path
+    int indexOfPath(const FileSystemPathComboEdit *fsPathEdit, const QString &savePath)
+    {
+        const QDir saveDir {savePath};
+        for (int i = 0; i < fsPathEdit->count(); ++i)
+        {
+            if (QDir(fsPathEdit->item(i)) == saveDir)
+                return i;
+        }
+        return -1;
+    }
+
+    void setPath(FileSystemPathComboEdit *fsPathEdit, const QString &newPath)
+    {
+        int existingIndex = indexOfPath(fsPathEdit, newPath);
+        if (existingIndex < 0)
+        {
+            // New path, prepend to combo box
+            fsPathEdit->insertItem(0, newPath);
+            existingIndex = 0;
+        }
+
+        fsPathEdit->setCurrentIndex(existingIndex);
+    }
+
+    void updatePathHistory(const QString &settingsKey, const QString &path, const int maxLength)
+    {
+        // Add last used save path to the front of history
+
+        auto pathList = settings()->loadValue<QStringList>(settingsKey);
+        QVector<QDir> dirs;
+        for (const QString &path : asConst(pathList))
+            dirs.append(QDir(path));
+
+        const QDir pushedDir {path};
+        const int selectedSavePathIndex = dirs.indexOf(pushedDir);
```

Comment:
seems `dirs` is redundant. you could just search and record the value with a loop.

Reply:
> seems `dirs` is redundant. you could just search and record the value with a loop.

It is previously existing logic but I anyway changed it.